orbit_determination/test_ekf.py

```diff
@@ -0,0 +1,156 @@
+"""
+Testing the EKF class.
+"""
+
+import pickle
+from time import time
+
+import brahe
+from brahe.epoch import Epoch
+import matplotlib.pyplot as plt
+import numpy as np
+import quaternion
+
+from dynamics.orbital_dynamics import f  # , f_jac
+from orbit_determination.ekf import EKF
+from orbit_determination.landmark_bearing_sensors import (
+    GroundTruthLandmarkBearingSensor,
+    SimulatedMLLandmarkBearingSensor,
+)
+from orbit_determination.od_simulation_data_manager import ODSimulationDataManager
+from sensors.bias import BiasParams
+from sensors.imu import IMU, IMUNoiseParams
+from sensors.sensor import SensorNoiseParams
+from utils.config_utils import load_config
+from utils.orbit_utils import get_sso_orbit_state  # , is_over_daytime
+
+
+def imu_init(dt: float) -> IMU:
+    """
+    Initializes the IMU.
+
+    :param dt: The time step for the simulation.
+
+    :return: The initialized IMU.
+    """
+    # Initialize the IMU
+    bias_params = BiasParams.get_random_params([0, 0], [1e-4, 1e-3])
+    sensor_noise_params_accel_x = SensorNoiseParams(bias_params, 5e-4, 5e-4)
+    sensor_noise_params_accel_y = SensorNoiseParams(bias_params, 5e-4, 5e-4)
+    sensor_noise_params_accel_z = SensorNoiseParams(bias_params, 5e-4, 5e-4)
+    sensor_noise_params_accel = [
+        sensor_noise_params_accel_x,
+        sensor_noise_params_accel_y,
+        sensor_noise_params_accel_z,
+    ]
+
+    sensor_noise_params_gyro_x = SensorNoiseParams(bias_params, 5e-4, 5e-4)
+    sensor_noise_params_gyro_y = SensorNoiseParams(bias_params, 5e-4, 5e-4)
+    sensor_noise_params_gyro_z = SensorNoiseParams(bias_params, 5e-4, 5e-4)
+    sensor_noise_params_gyro = [
+        sensor_noise_params_gyro_x,
+        sensor_noise_params_gyro_y,
+        sensor_noise_params_gyro_z,
+    ]
+
+    imu_noise_params = IMUNoiseParams(
+        gyro_params=sensor_noise_params_gyro, accel_params=sensor_noise_params_accel
+    )
+    imu = IMU(dt, imu_noise_params)
+
+    return imu
+
+
+def run_simulation() -> None:
+    """
+    Run the simulation.
+
+    :return: None
+    """
+
+    config = load_config()
+
+    config["solver"]["world_update_rate"] = 1 / 60  # Hz
+    config["mission"]["duration"] = 3 * 90 * 60  # s, roughly 1 orbit
+
+    dt = 1 / config["solver"]["world_update_rate"]
+    starting_epoch = Epoch(*brahe.time.mjd_to_caldate(config["mission"]["start_date"]))
+    N = int(np.ceil(config["mission"]["duration"] / dt))  # number of time steps in the simulation
+
+    landmark_bearing_sensor = GroundTruthLandmarkBearingSensor(config)
+    data_manager = ODSimulationDataManager(starting_epoch, dt)
+
+    initial_state = get_sso_orbit_state(starting_epoch, 0, -73, 600e3, northwards=True)
+    init_rot = quaternion.as_rotation_matrix(quaternion.as_quat_array(np.array([1, 0, 0, 0])))
+
+    data_manager.push_next_state(
+        np.expand_dims(initial_state, axis=0), np.expand_dims(init_rot, axis=0)
```

Comment:
No need to use `np.expand_dims`, `push_next_state(initial_state, init_rot)` should work now.
```suggestion
        initial_state, init_rot
```

Reply:
Yeah, that's why I had put `[np.newaxis, ...]` in the implementation of `push_next_state` (I think it got removed in this PR). My general thinking is that this is an implementation detail and you shouldn't have to worry about it from the calling code.